doc/Gemm.md

```diff
@@ -0,0 +1,405 @@
+
+- Work on or create a new `GEMM` kernel
+- Work with the `GEMM` dispatch system.
+- Add or modify `GEMM` configurations for different backends.
+
+Please note that while this document primarily refers to `GEMM` and `Blas3` operations many of the same concepts around source code generation apply to other operations and blas levels.
+
+# Contents
+
+- [**GEMM**](#gemm)
+
+  - [What is GEMM?](#what-is-gemm)
+  - [SYCL-BLAS Gemm Kernels](#sycl-blas-gemm-kernels)
+  - [Relevant CMake Variables](#relevant-cmake-variables)
+  - [Kernel Structure](#kernel-structure)
+  - [Vectorized Loading/Storing](#vectorized-loading/storing)
+  - [Batched Gemm](#batched-gemm)
+
+- [**GEMM Dispatch**](#gemm-dispatch)
+
+  - [GEMM Backends](#gemm-backends)
+  - [GEMM Launcher](#gemm-launcher)
+  - [Source Code Generation](#source-code-generation)
+
+- [**GEMM Configurations**](#gemm-configurations)
+
+  - [Backend Configurations](#backend-configurations)
+  - [CMake Configurations](#cmake-configurations)
+
+- [**Common Tasks**](#common-tasks)
+
+  - [Adding a new configuration](#adding-a-new-configuration)
+  - [Adding a new kernel](#adding-a-new-kernel)
+
+# GEMM
+
+## What is GEMM?
+
+`GEMM` stands for General Matrix Multiplication and solves the equation of the form:
+
+```
+C = alpha * A * B + beta * C
+where A, B and C are matrices and alpha and beta are scalars.
+```
+
+## SYCL-BLAS GEMM Kernels
+
+`SYCL-BLAS` currently contains the following `GEMM` kernels in <src/operations/blas3/>:
+
+- `gemm_ref.hpp` - A naive, reference implementation of `GEMM` with no optimizations.
+
+- `gemm_partial_local.hpp` - Used for tall, skinny `GEMM` optimizations.
+
+- `gemm_local.hpp` - Uses local memory for increased performance on platforms that have it. Supports full vectorization.
+
+- `gemm_no_local_partial_vec.hpp` - Doesn't use local memory. Only supports partial vectorization (can only vectorize in some specific cases).
+
+- `gemm_no_local_full_vec.hpp` - Doesn't use local memory. Supports full vectorization.
+
+- `gemm_interleaved.hpp` - An alternative approach to batched `GEMM` calculations where the inputs are interleaved in contiguous memory. Uses no local memory and corresponds to HWN data layout (NWH in column major, which is what `SYCL-BLAS` uses).
+
+## Relevant CMake Variables
+
+There are several CMake variables which are specific to `GEMM` :
+
+- `NAIVE_GEMM` (Default: `OFF`) - Forces the use of the naive, reference `GEMM` implementation.
+- `GEMM_VECTORIZATION_SUPPORT` (Default: `OFF`) - Enables vectorization within the `GEMM` kernels. If `OFF` it is equivalent to passing `1` for the vector size to the `Gemm` launcher.
+- `GEMM_TALL_SKINNY_SUPPORT` (Default: `ON`) - Enables optimizations for tall, skinny matrices. Not used on all targets.
+- `BLAS_MODEL_OPTIMIZATION` - Passing a machine learning model name here (`VGG_16` or `RESNET_50`) enables optimizations for the `Gemm` sizes used in these models. Only applies to the `ARM_GPU` target.
```

Comment:
```suggestion
- `BLAS_MODEL_OPTIMIZATION` - Passing a machine learning model name here (`VGG_16` or `RESNET_50`) enables optimizations for the `GEMM` sizes used in these models. Only applies to the `ARM_GPU` target.
```

I think you use the all caps `GEMM` more often than the lowecase one when refering to the operation.

Reply:
I've done a pass on this and it should be more consistent now, `Gemm` should only be used when explicitly referring to the class and `GEMM` in all other cases.